### ISSUE TITLE
Ability to send files

### ISSUE THREAD

**Issue by deavmi** [Nutomic/ensichat#21]
The ability to send files through the mesh would be awesome. And maybe even (sign the files, I think though that signatures are already in the protocol for text messages in EnsiChat if I am correct.

**Comment by Nutomic** [comment]
This is a bit complicated, as we'll have to split the files in chunks. And we need a better routing protocol, as flood routing will probably kill all connections when a lot of data is transferred.

**Comment by deavmi** [comment]
That's true. There are obviously more efficient routing protocols, I do not know what they r called, but I think one of them is dynamic routing. Anyway, I agree that the routing protocol should later be updated.

**Comment by deavmi** [comment]
Is this because antennas connect to devices one by one and sweep through them by connecting and disconnecting to the devices.

**Comment by Nutomic** [comment]
No, it's because messages are sent to every connected device, on the entire network. Bluetooth has concurrent connections afaik. So it's just a problem of the app code.

**Comment by deavmi** [comment]
Oh okay.

**Comment by deavmi** [comment]
I was wondering, Canada Bluetooth device send data to a device right at the exact same time as another? Physically that really doesn't sound right?

**Comment by Nutomic** [comment]
I don't know how this is on a physical level, but it should work with the Android API.

**Comment by deavmi** [comment]
Yeah. Okay. Just saying the physically realm. Nothing happens at once, there is always the first command. Parallel processing does not exist if the commands to do the processing cannot be executed at the exactly the same time.

**Comment by Nutomic** [comment]
That's nothing I'm really concerned with :p

**Comment by deavmi** [comment]
Yeah. Just saying my thoughts. But as long as the android api acts like so then it works.
